clang/lib/StaticAnalyzer/Checkers/StreamChecker.cpp

```diff
@@ -544,6 +545,21 @@ const ExplodedNode *StreamChecker::getAcquisitionSite(const ExplodedNode *N,
   return nullptr;
 }
 
+static ProgramStateRef
+escapeArgs(ProgramStateRef State, CheckerContext &C, const CallEvent &Call,
+           const SmallVector<unsigned int> &EscapingArgs) {
+  const auto *CE = Call.getOriginExpr();
+
+  SmallVector<SVal> EscapingVals;
+  EscapingVals.reserve(EscapingArgs.size());
+  for (auto EscArgIdx : EscapingArgs)
+    EscapingVals.push_back(Call.getArgSVal(EscArgIdx));
+  State = State->invalidateRegions(EscapingVals, CE, C.blockCount(),
+                                   C.getLocationContext(),
+                                   /*CausesPointerEscape=*/false);
```

Comment:
```suggestion
  State = State->invalidateRegions(EscapingVals, CE, C.blockCount(),
                                   C.getLocationContext(),
                                   /*CausesPointerEscape=*/false, /*InvalidatedSymbols=*/nullptr, &Call);
```

I can't recall now what difference it makes to pass the `Call` to this API, but given that we have one, why not?

Reply:
If this change is made, a error shows up in **stream-errno.c** line 190 (unexpected warning: FALSE).
```
  int Ret = fgetpos(F, &Pos);
  if (Ret)
    clang_analyzer_eval(errno != 0); // expected-warning{{TRUE}}
  else
    clang_analyzer_eval(errno == 0); // expected-warning{{TRUE}} we get FALSE too
```
Probably the `errno` is invalidated somehow, this is likely to be caused by the `Call` argument. Without these additional arguments to `invalidateRegions` the test passes.

Reply:
Yup, have a look at `RegionStoreManager::invalidateRegions`. Depending on the nature of the call, invalidates system or every global variables.
We don't really care about it, so let's just not pass the `Call` here.